File: 3105. Longest Strictly Increasing or Strictly Decreasing Subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        answer, inc, dec = 1, 1, 1
        for i in range(1, len(nums)):
            inc = inc+1 if nums[i] > nums[i-1] else 1
            dec = dec+1 if nums[i] < nums[i-1] else 1
            answer = max(answer, inc, dec)
        return answer

File: 3105. Longest Strictly Increasing or Strictly Decreasing Subarray.go
func longestMonotonicSubarray(nums []int) int {
	answer, inc, dec := 1, 1, 1
	for i := 1; i < len(nums); i++ {
		if nums[i] > nums[i-1] {
			inc += 1
			dec = 1
		} else if nums[i] < nums[i-1] {
			dec += 1
			inc = 1
		} else {
			inc = 1
			dec = 1
		}
		answer = max(answer, inc, dec)
	}
	return answer
}